src/credentials/sharedCredentialsValidation.ts

```diff
@@ -0,0 +1,94 @@
+/*!
```

Comment:
Similar to https://github.com/aws/aws-toolkit-vscode/blob/master/src/shared/sam/debugger/awsSamDebugConfigurationValidator.ts this file has a lot of code for basic validation.

I wonder if it could be de-duplicated or generalized. Not a blocker for this PR though. Maybe this is another case where something like [zod](https://github.com/aws/aws-toolkit-vscode/issues/3368) could be useful.

Reply:
I agree, and with the current approach we can easily modify to use a better approach